controller/modelmanager/createmodel.go

```diff
@@ -69,6 +70,27 @@ func (c ModelConfigCreator) NewModelConfig(
 		attrs[config.UUIDKey] = uuid.String()
 	}
 	attrs[config.TypeKey] = cloud.Type
+
+	// We need to get the system-identity public key to be added to the
+	// newly created model config at AuthorizedKeys.
+	// First, we take the controller model's AuthorizedKeys one by one and
+	// try to find the one corresponding to system-identity.
+	for _, key := range ssh.SplitAuthorisedKeys(base.AuthorizedKeys()) {
+		parsedKey, err := ssh.ParseAuthorisedKey(key)
+		if err != nil {
+			logger.Tracef("error parsing controller authorized key %s", key)
+			continue
+		}
+		// system-identity public key must be commented with
+		// Juju:system-identity
```

Comment:
I think `Juju:system-identity` is the wrong string here, at least your test cases says `juju-system-key`, and

Reply:
you are right, fixed!